Build Error with buffer_base.cpp

# System Information
`Arch Linux`
`glibc 2.36`
`clang 14.0.6`
`gcc 12.2.0`
`gcc-libs 12.2.0`

# Build flags
```
CFLAGS="-march=znver3 -O3 -pipe -fno-plt -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security \
        -fstack-clash-protection -fcf-protection -flto=thin"
CXXFLAGS="$CFLAGS -Wp,-D_GLIBCXX_ASSERTIONS"
CC="clang"
CCX="clang++"
```

## CMAKE Flags
```
    cmake .. \
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DCMAKE_C_COMPILER=clang \
      -DCMAKE_CXX_COMPILER=clang++ \
      -DCMAKE_C_FLAGS="$CFLAGS -flto=thin" \
      -DCMAKE_CXX_FLAGS="$CXXFLAGS -flto=thin" \
      -DCMAKE_C_COMPILER_LAUNCHER="ccache" \
      -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" \
      -DCMAKE_BUILD_TYPE=RelWithDebInfo \
      -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
      -DYUZU_USE_QT_WEB_ENGINE=ON \
      -DYUZU_USE_EXTERNAL_SDL2=OFF \
      -DUSE_DISCORD_PRESENCE=ON \
      -DENABLE_QT_TRANSLATION=ON \
      -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON \
      -DBUILD_REPOSITORY=yuzu-emu/yuzu-mainline \
      -DBUILD_TAG=${pkgver} \
      -DTITLE_BAR_FORMAT_IDLE="yuzu | ${pkgver} {}" \
      -DTITLE_BAR_FORMAT_RUNNING="yuzu | ${pkgver} | {}" \
      -DDYNARMIC_NO_BUNDLED_ROBIN_MAP=ON \
      -DYUZU_USE_BUNDLED_OPUS=OFF \
      -DYUZU_USE_BUNDLED_FFMPEG=OFF \
      -DYUZU_USE_BUNDLED_LIBUSB=OFF \
      -DYUZU_USE_BUNDLED_QT=OFF
```

# Error
## Clang Error Information
```
[ 90%] Building CXX object src/tests/CMakeFiles/tests.dir/video_core/buffer_base.cpp.o
[ 90%] Built target yuzu_autogen
Consolidate compiler generated dependencies of target yuzu
[ 99%] Built target yuzu
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:25: error: loop variable '[index, value]' creates a copy from type 'const std::__detail::_Node_const_iterator<std::pair<const unsigned long, int>, false, false>::value_type' (aka 'const std::pair<const unsigned long, int>') [-Werror,-Wrange-loop-construct]
        for (const auto [index, value] : page_table) {
                        ^
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:14: note: use reference type 'const std::__detail::_Node_const_iterator<std::pair<const unsigned long, int>, false, false>::value_type &' (aka 'const std::pair<const unsigned long, int> &') to prevent copying
        for (const auto [index, value] : page_table) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                        &
1 error generated.
make[2]: *** [src/tests/CMakeFiles/tests.dir/build.make:216: src/tests/CMakeFiles/tests.dir/video_core/buffer_base.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1561: src/tests/CMakeFiles/tests.dir/all] Error 2
```
## GCC Error Information
```
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp: In member function ‘unsigned int {anonymous}::RasterizerInterface::Count() const’:
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:25: error: loop variable ‘<structured bindings>’ creates a copy from type ‘const std::pair<const long unsigned int, int>’ [-Werror=range-loop-construct]
   47 |         for (const auto [index, value] : page_table) {
      |                         ^~~~~~~~~~~~~~
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:25: note: use reference type to prevent copying
   47 |         for (const auto [index, value] : page_table) {
      |                         ^~~~~~~~~~~~~~
      |                         &
```
# Extra Info
As the world's most extreme yuzu fanatic (you can't disprove this, I will fight to the death), I am reporting this bug as it is my way of contributing to this great free-and-open-source project.

Wait, I am still getting the error even with `-DBUILD_TESTS=0`
```
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:25: error: loop variable '[index, value]' creates a copy from type 'const std::__detail::_Node_const_iterator<std::pair<const unsigned long, int>, false, false>::value_type' (aka 'const std::pair<const unsigned long, int>') [-Werror,-Wrange-loop-construct]
        for (const auto [index, value] : page_table) {
                        ^
/home/hurricane/yuzu-mainline-git/src/yuzu/src/tests/video_core/buffer_base.cpp:47:14: note: use reference type 'const std::__detail::_Node_const_iterator<std::pair<const unsigned long, int>, false, false>::value_type &' (aka 'const std::pair<const unsigned long, int> &') to prevent copying
        for (const auto [index, value] : page_table) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                        &
1 error generated.
```

--- a/src/tests/video_core/buffer_base.cpp
+++ b/src/tests/video_core/buffer_base.cpp
@@ -44,7 +44,7 @@ public:
 
     [[nodiscard]] unsigned Count() const noexcept {
         unsigned count = 0;
-        for (const auto [index, value] : page_table) {
+        for (const auto& [index, value] : page_table) {
             count += value;
         }
         return count;

